libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtilsTest.kt

```diff
@@ -194,37 +193,32 @@ class MapboxRouteLineUtilsTest {
 
     @Test
     fun getRestrictedSectionExpressionData() {
-        val route = loadRoute("route-with-restrictions.json")
-
-        val result = MapboxRouteLineUtils.extractRouteDataWithTrafficAndRoadClassDeDuped(
-            route,
-            MapboxRouteLineUtils.getRouteLegTrafficCongestionProvider
+        val route = loadRoute("route-with-restrictions.json").toNavigationRoute(
+            RouterOrigin.Offboard
         )
 
-        assertEquals(14, result.size)
-        assertTrue(result[4].isInRestrictedSection)
-        assertFalse(result[5].isInRestrictedSection)
-        assertTrue(result[6].isInRestrictedSection)
+        val result = MapboxRouteLineUtils.extractRouteRestrictionData(route)
+
+        assertEquals(4, result.size)
+        assertTrue(result[1].isInRestrictedSection)
+        assertTrue(result[2].isInRestrictedSection)
+        assertFalse(result[3].isInRestrictedSection)
     }
 
     @Test
     fun getRestrictedLineExpression() {
-        val expectedExpression = "[step, [line-progress], [rgba, 0.0, 0.0, 0.0, 0.0], " +
-            "0.2, [rgba, 0.0, 0.0, 0.0, 0.0], 0.44865144220494346, " +
-            "[rgba, 255.0, 255.0, 255.0, 1.0], 0.468779750455607, [rgba, 0.0, 0.0, 0.0, 0.0]," +
-            " 0.5032854217424586, [rgba, 255.0, 255.0, 255.0, 1.0], 0.5207714038134984, " +
-            "[rgba, 0.0, 0.0, 0.0, 0.0]]"
-
-        val route = loadRoute("route-with-restrictions.json")
-        val expData = MapboxRouteLineUtils.extractRouteData(
-            route,
-            MapboxRouteLineUtils.getRouteLegTrafficCongestionProvider
+        val expectedExpression = "[step, [line-progress], [rgba, 0.0, 0.0, 0.0, 0.0], 0.2, " +
+            "[rgba, 0.0, 0.0, 0.0, 0.0], 0.4476941554901612, [rgba, 0.0, 255.0, 255.0, 1.0], " +
```

Comment:
Here we have step from black to black, why is that? Shouldn't this be filtered out?

Reply:
`[step, [line-progress], [rgba, 0.0, 0.0, 0.0, 0.0]` sets up the Expression with a default color which in this case is transparent.  After this hardcoded syntax, the collection of data items is processed.  The loop processing the data items doesn't check the default gradient color. I can adjust it.

Reply:
`[step, [line-progress], [rgba, 0.0, 0.0, 0.0, 0.0]` is a hardcoded syntax for setting up the gradient expression.  In this case the gradient is given a default color of transparent. After this, stops are added to the gradient based on the data items.  There has to be at least one stop in the `Expression`.  If that method gets called with an empty list of data items, internally it will still create one stop in the `Expression`. So if the first (or only) item has the same (transparent) color and it got filtered out leaving no stops for the `Expression` the syntax would be incorrect and maps throws an exception.

Reply:
Got it :+1: